packages/connect-core/src/utils/metadata.ts

```diff
@@ -39,3 +48,29 @@ export async function resolveMetadata(
   if (contentUri) return fetchMetadata(fileName, contentUri)
   return {}
 }
+
+export async function resolveManifest(
+  data: AppData | RepoData
+): Promise<AragonManifest> {
+  return resolveMetadata(
+    'manifest.json',
+    data.contentUri || undefined,
+    data.manifest
+  )
+}
+
+export async function resolveArtifact(
+  data: AppData | RoleData
+): Promise<AragonArtifact> {
+  if (hasAppInfo(data.appId, 'apm')) {
+    return getApmInternalAppInfo(data.appId)
+  } else if (hasAppInfo(data.appId, 'aragon')) {
```

Comment:
Now that we are returning from a function, we could get rid of the `else if` here, and the `else` below.

Reply:
Nice, doing that change.